libtiny_tui/src/msg_area/line.rs

```diff
@@ -200,65 +187,41 @@ impl Line {
         pos_y: i32,
         first_line: i32,
         height: i32,
-        width: i32,
     ) {
         let mut col = pos_x;
-        let mut line = 0;
-
-        let mut next_split_idx: usize = 0;
-
-        let mut char_idx: i32 = 0;
+        let mut line_num = 0;
+        let mut char_idx = 0;
+        let mut split_indices_iter = self.line_data.get_splits().iter().copied().peekable();
 
-        let last_seg: [&Seg; 1] = [&self.current_seg];
-        for seg in self.segs.iter().chain(last_seg.iter().copied()) {
+        let last_seg: [&StyledString; 1] = [&self.current_seg];
+        for seg in self.string_segments.iter().chain(last_seg.iter().copied()) {
```

Comment:
No need to copy `last_seg` here

```suggestion
        for seg in self.string_segments.iter().chain(::std::iter::once(&self.current_seg)) {
```

Tbh I don't understand why `chain` here needs an iterator that owns the data, but it's not too important.